### PR TITLE
[JT-47] /api/v2/categories documentation

### PR DESCRIPTION
**Pull request by lbenamer**
[Story](https://partoo.atlassian.net/secure/RapidBoard.jspa?rapidView=27&projectKey=JT&modal=detail&selectedIssue=JT-47)

* endpoint : https://preprod.partoo.co/api/v2/categories

Screen Shot 2020-06-16 at 11 49 17

### CODE REVIEW

File: rest_api/v2/components/index.yml [hunk 1]
```diff
@@ -133,6 +133,57 @@ schemas:
...
+        description: Country code
+        example: 'FR'
+  CategorySqlNames:
+    type: array
+    description: Category name translated for available countries
+    items:
+      $ref: '#/schemas/CategorySqlName'
+  CategorySqlLegacyIds:
+    type: array
+    description: old related partoo categories id
```
Comment (ThibC) [inline rest_api/v2/components/index.yml:162]: To make it consistent with other `description` ;) 
```suggestion
    description: Old related partoo categories id
```

File: rest_api/v2/components/index.yml [hunk 2]
```diff
@@ -133,6 +133,57 @@ schemas:
...
+    example: ['5109c3a237zdc544d8e267378', 'bf5179d3a237zdc576d8e2673']
+  CategorySqlState:
+    type: string
+    description: state of category ( OK or DEPRECATED )
```
Comment (ThibC) [inline rest_api/v2/components/index.yml:166]:
```suggestion
    description: State of category ( OK or DEPRECATED )
```

File: rest_api/v2/components/index.yml [hunk 3]
```diff
@@ -1613,6 +1664,28 @@ parameters:
...
+    in: query
+    name: state
+    required: false
+    schema:
+      type: string
+      enum: ['OK', 'DEPRECATED']
+  query_category_block_push_long:
+    in: query
+    name: block_push_long_description_on_gmb
+    description: Long description can't be set on GMB if a business own a category with this flag set to True
```
Comment (ThibC) [inline rest_api/v2/components/index.yml:1685]: I am not sure we should describe what does `block_push_long_description_on_gmb` means here :)

```suggestion
    description: Filter on `block_push_long_description_on_gmb` param 
```

File: rest_api/v2/components/index.yml [hunk 4]
```diff
@@ -133,6 +133,57 @@ schemas:
...
+    type: array
+    description: old related partoo categories id
+    example: ['5109c3a237zdc544d8e267378', 'bf5179d3a237zdc576d8e2673']
+  CategorySqlState:
+    type: string
+    description: state of category ( OK or DEPRECATED )
+    example: 'OK'
+  CategorySqlBlockPushLong:
+    type: boolean
+    description:  Long description can't be set on GMB if a business own a category with this flag set to True
```
Comment (ThibC) [inline rest_api/v2/components/index.yml:170]: Maybe it is clearer written like this :)

```suggestion
    description:  Boolean indicating if the business long description can be pushed to GMB when the business belongs to this category
```

File: rest_api/v2/components/index.yml (hunk 2 above)
Comment (lbenamer) [reply rest_api/v2/components/index.yml:166]: 👍

File: rest_api/v2/components/index.yml (hunk 1 above)
Comment (lbenamer) [reply rest_api/v2/components/index.yml:162]: 👍

File: rest_api/v2/components/index.yml (hunk 4 above)
Comment (lbenamer) [reply rest_api/v2/components/index.yml:170]: very more elegant :D

File: rest_api/v2/components/index.yml (hunk 3 above)
Comment (lbenamer) [reply rest_api/v2/components/index.yml:1685]: I was thinking it was not so bad to explain what this attribute means in a way this information can be useful for clients

Comment (lbenamer) [reply rest_api/v2/components/index.yml:1685]: I was thinking it was not so bad to explain what this attribute means in a way this information can be useful for clients

Comment (ThibC) [reply rest_api/v2/components/index.yml:1685]:
```suggestion
    description: Filter on `block_push_long_description_on_gmb` param (boolean indicating if the business long description can be pushed to GMB when the business belongs to this category)
```

Comment (ThibC) [reply rest_api/v2/components/index.yml:1685]: Comme ça?

Comment (lbenamer) [reply rest_api/v2/components/index.yml:1685]: lets go 😄

**Pull request merged**